k8s/plugin/src/resources/upgrade.rs

```diff
@@ -54,187 +62,371 @@ impl UpgradeResources {
         })
     }
 
-    /// Install the upgrade resources
-    pub async fn install(ns: &str) {
-        match UpgradeResources::new(ns).await {
-            Ok(uo) => {
-                let pp = PostParams::default();
-                // Create a service account
-                match uo.get_service_account().await {
-                    Ok(sa) => {
-                        if sa.iter().count() == 0 {
-                            let ns = Some(ns.to_string());
-                            let service_account = objects::upgrade_operator_service_account(
-                                ns,
-                                uo.release_name.clone(),
-                            );
-                            let pp = PostParams::default();
-                            match uo.service_account.create(&pp, &service_account).await {
-                                Ok(sa) => {
-                                    println!(
-                                        "Service Account : {} created in namespace : {}.",
-                                        sa.metadata.name.unwrap(),
-                                        sa.metadata.namespace.unwrap(),
-                                    );
-                                }
-                                Err(e) => {
-                                    println!("Failed in creating service account  {:?}", e);
-                                    tokio::time::sleep(Duration::from_secs(1)).await;
-                                    std::process::exit(1);
-                                }
-                            }
-                        } else {
-                            println!(
-                                "Service Account : {} in namespace : {} already exist.",
-                                sa.items[0].metadata.name.as_ref().unwrap(),
-                                sa.items[0].metadata.namespace.as_ref().unwrap(),
-                            );
+    /// Create/Delete ServiceAction
```

Comment:
not for this PR, but please move all this code to a library as this is becoming too unwieldy to be in the plugin, instead should be imported from a library.